app/views/questionnaire.py

```diff
@@ -330,13 +332,25 @@ def get_view_submission(schema, eq_id, form_type):  # pylint: disable=unused-arg
     return redirect(url_for('post_submission.get_thank_you', eq_id=eq_id, form_type=form_type))
 
 
-def _set_started_at_metadata_if_required(form, metadata):
-    questionnaire_store = get_questionnaire_store(current_user.user_id, current_user.user_ik)
-    if not questionnaire_store.answer_store.answers and len(form.data) > 1:
-        started_at = datetime.now(timezone.utc).isoformat()
-        logger.info('first answer about to be stored. writing started_at time to metadata',
+def _set_started_at_metadata_if_required(form, collection_metadata, metadata):
+    if not collection_metadata.get('started_at') and form.data:
+        started_at = datetime.utcnow().isoformat()
+
+        # To avoid loss of in-flight data while collection_metadata is deployed.
+        # After deployment of collection metdata, remove from here to the next comment.
```

Comment:
Loving this.  Decent instructions on how to remove code after it's not useful anymore is welcome to see!